src/datalog/data_access_analysis.dl

```diff
@@ -343,5 +343,41 @@ propagated_data_access(NextEA,Mult,EA_ref):-
         data_access_pattern(EA_ref,S,Mult,From)
     ).
 
+// last data_access_pattern_candidate
+.decl last_data_access_candidate(EA:address,Where:address)
+.output last_data_access_candidate
+
+last_data_access_candidate(EA,Where):-
+    data_access_pattern_candidate(EA,_,_,_),
+    Where = max Where : {data_access_pattern_candidate(Where,_,_,_), Where < EA}.
+
+// adjacent non collision data accesses
+.decl non_collision_access(EA:address,EA_ref:address)
+.output non_collision_access
+
+non_collision_access(EA,EA):-
+    data_access_pattern_candidate(EA,_,Mult,_),
+    Mult > 0.
+
+// connecting adjacent data accesses when they have the same multiplier
+// and they are not colliding
+non_collision_access(EA_prop,EA_access):-
+    last_data_access_candidate(EA_prop,EA_access),
+    data_access_pattern_candidate(EA_prop,_,Mult,_),
+    data_access_pattern_candidate(EA_access,_,Mult,_), Mult > 0,
+    (EA_prop - EA_access) % as(Mult,unsigned) = 0.
+
+// If a data access at NextEA lies in the middle of [EA, EA + Mult], it's 
+// still not colliding if the NextEA >= EA + Size.
+non_collision_access(NextEA,EA_ref):-
+    non_collision_access(EA,EA_ref),
+    NextEelementEA = EA+ as(Mult,unsigned),
+    // There is a data access in between
+    last_data_access_candidate(NextEA, EA), 
+    NextEelementEA > NextEA,
+    //But it is "synchronized": It has the same multiplier and it does not collide
+    data_access_pattern_candidate(EA,Size,Mult,_),
+    EA+Size <= NextEA.
```

Comment:
Given this rule, I think it is possible that you end up with multiple references for a same address.
For example, if we have accesses at addresses `0`, `2`, and `4` with multiplier `10` (with e.g. size 1).  I think you will get the tuples:
```
non_collision_access(4,2)
non_collision_access(4,0)
```
- The first one would be generated by matching: `EA=2` `EA_ref=2` and `NextEA=4`.
- The second one would be generated in two steps: non_collision_access(2,0) would be generated first, and then you would match `EA=2`, `EA_ref=0` and `NextEA=4`.

This could lead to multiple `moved_label` predicates for a single symbolic operand, which in turn can lead to bugs hard to debug. Maybe there should be a subsequent refinement to choose the minimum reference for a given address (so in this example we keep `(4,0)` rather than `(4,2)`).